src/runtime-rs/crates/hypervisor/src/firecracker/inner.rs

```diff
@@ -0,0 +1,172 @@
+//Copyright (c) 2019-2022 Alibaba Cloud
+//Copyright (c) 2023 Nubificus Ltd
+//
+//SPDX-License-Identifier: Apache-2.0
+
+use crate::firecracker::utils::get_api_socket_path;
+use crate::HypervisorState;
+use crate::HYPERVISOR_FIRECRACKER;
+use crate::{device::DeviceType, VmmState};
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use hyper::Client;
+use hyperlocal::{UnixClientExt, UnixConnector};
+use kata_types::{
+    capabilities::{Capabilities, CapabilityBits},
+    config::hypervisor::Hypervisor as HypervisorConfig,
+};
+use nix::sched::{setns, CloneFlags};
+use persist::sandbox_persist::Persist;
+use std::os::unix::io::AsRawFd;
+use tokio::process::Command;
+
+unsafe impl Send for FcInner {}
+unsafe impl Sync for FcInner {}
+
+#[derive(Debug)]
+pub struct FcInner {
+    pub(crate) id: String,
+    pub(crate) asock_path: String,
+    pub(crate) state: VmmState,
+    pub(crate) config: HypervisorConfig,
+    pub(crate) pid: Option<u32>,
+    pub(crate) vm_path: String,
+    pub(crate) netns: Option<String>,
+    pub(crate) client: Client<UnixConnector>,
+    pub(crate) jailer_root: String,
+    pub(crate) jailed: bool,
+    pub(crate) pending_devices: Vec<DeviceType>,
+    pub(crate) capabilities: Capabilities,
+}
+
+impl FcInner {
+    pub fn new() -> FcInner {
+        let mut capabilities = Capabilities::new();
+        capabilities.set(CapabilityBits::BlockDeviceSupport);
+        FcInner {
+            id: String::default(),
+            asock_path: String::default(),
+            state: VmmState::NotReady,
+            config: Default::default(),
+            pid: None,
+            netns: None,
+            vm_path: String::default(),
+            client: Client::unix(),
+            jailer_root: String::default(),
+            jailed: false,
+            pending_devices: vec![],
+            capabilities,
+        }
+    }
+
+    pub(crate) async fn prepare_vmm(&mut self, netns: Option<String>) -> Result<()> {
+        let mut cmd: Command;
+        self.netns = netns.clone();
+        match !self.jailed {
```

Comment:
Just playing a linter here ;-) but wouldn't this be a bit easier to read if it was `match self.jailed` and the arms were swapped?

Reply:
I mean sure, it was just that the jailer function came afterwards and that is why it is ordered like that, if the reviewers have a problem it will be swapped no problem.

Reply:
ack!